gdal/gcore/gdalrasterband.cpp

```diff
@@ -5219,6 +5279,34 @@ GDALRasterBand::ComputeStatistics( int bApproxOK,
         SetStatistics( dfMin, dfMax, dfMean, dfStdDev );
     }
 
+    if( nValidCount == 0 )
```

Comment:
as I see the below snippet is found before, can we factor it in a helper function ?